pkg/cmd/addon/index.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package addon
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+
+	"github.com/apecloud/kbcli/pkg/printer"
+	"github.com/apecloud/kbcli/pkg/types"
+	"github.com/apecloud/kbcli/pkg/util"
+)
+
+type index struct {
+	name string
+	url  string
+}
+
+func newIndexListCmd(streams genericiooptions.IOStreams) *cobra.Command {
+	indexListCmd := &cobra.Command{Use: "list",
+		Short: "List addon indexes",
+		Long: `Print a list of addon indexes.
+
+This command prints a list of addon indexes. It shows the name and the remote URL for
+each addon index in table format.`,
+		Args: cobra.NoArgs,
+		Run: func(_ *cobra.Command, _ []string) {
+			util.CheckErr(listIndexes(streams.Out))
+		},
+	}
+	return indexListCmd
+}
+
+func newIndexAddCmd() *cobra.Command {
+	indexAddCmd := &cobra.Command{
+		Use:     "add",
+		Short:   "Add a new addon index",
+		Long:    "Configure a new index to install KubeBlocks addon from.",
+		Example: "kbcli index add kubeblocks " + types.DefaultAddonIndexURL,
+		Args:    cobra.ExactArgs(2),
+		Run: func(_ *cobra.Command, args []string) {
+			util.CheckErr(addIndex(args))
+			fmt.Printf("You have added a new index from %q\n", args[1])
+		},
+	}
+
+	defaultIndexCmd := &cobra.Command{
```

Comment:
Remove this command, and like https://github.com/apecloud/kbcli/blob/29ffeb26dd7cc2707c1c5024464c054000fcb4f5/pkg/cmd/plugin/plugin.go#L61-L61, when run corresponding subcommand, check index and init the default index if not set.